src/main/java/io/cdap/plugin/google/drive/source/GoogleDriveSourceConfig.java

```diff
@@ -134,15 +232,103 @@ public GoogleDriveSourceConfig(String referenceName, @Nullable String fileMetada
     this.endDate = endDate;
   }
 
+  @Override
+  public void validate(FailureCollector collector) {
+    getSchema();
+    // Extra validation when structure schema is required
+  }
+
+  @Override
+  public String getPath() {
+    IdentifierType idType = getIdentifierType();
+    if (idType == IdentifierType.FILE_IDENTIFIER) {
+      return String.format("%s://%s%s/%s/%s", GOOGLE_DRIVE_SCHEMA, GOOGLE_DRIVE_AUTHORITY,
+        GOOGLE_DRIVE_FILE_PATH_PREFIX, getFileIdentifier(), GOOGLE_DRIVE_DEFAULT_FILENAME);
+    } else if (idType == IdentifierType.DIRECTORY_IDENTIFIER) {
+      return String.format("%s://%s%s/%s/", GOOGLE_DRIVE_SCHEMA, GOOGLE_DRIVE_AUTHORITY,
+        GOOGLE_DRIVE_FOLDER_PATH_PREFIX, getDirectoryIdentifier());
+    }
+    throw new IllegalArgumentException(String.format("Invalid identifier type '%s'. Expected one of: %s or %s.", idType,
+        IdentifierType.FILE_IDENTIFIER, IdentifierType.DIRECTORY_IDENTIFIER));
+  }
+
+  @Override
+  public String getPath(StageContext context) {
+    return getPath();
+  }
+
+  @Override
+  public String getFormatName() {
+    // need to do this for backwards compatibility, where the pre-packaged format names were case insensitive.
+    try {
+      FileFormat fileFormat = FileFormat.from(format, x -> true);
+      return fileFormat.name().toLowerCase();
+    } catch (IllegalArgumentException e) {
+      // ignore
+    }
+    return format;
+  }
+
+  @Nullable
+  @Override
+  public FileFormat getFormat() {
+    throw new UnsupportedOperationException("GDrive does not support: FileFormat getFormat() method");
+  }
+
+  @Nullable
+  @Override
+  public Pattern getFilePattern() {
+    return null;
```

Comment:
Does it makes sense to throw UnsupportedOperationException for this and other fields as well which are returning null?